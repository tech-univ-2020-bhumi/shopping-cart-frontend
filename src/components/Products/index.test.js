import React from 'react';
import { render } from '@testing-library/react';
import Product from './index';


describe('the product component', () => {
  it('should render correctly when cardStyle is normal', () => {
    const { asFragment } = render(<Product cardStyle="normal" />);

    expect(asFragment()).toMatchSnapshot();
  });
  it('should render correctly when cardStyle is sold-out', () => {
    const { asFragment } = render(<Product cardStyle="sold-out" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
