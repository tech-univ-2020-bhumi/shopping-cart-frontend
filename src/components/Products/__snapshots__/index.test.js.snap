// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the product cards should render correctly when cardStyle is normal 1`] = `
<DocumentFragment>
  <div
    class="product-card"
    data-testid="product"
  >
    <img />
    <p />
    <p>
      MRP  /
    </p>
    <div
      data-testid="counter-div"
    >
      <button
        data-testid="decrement"
        type="button"
      >
        -
      </button>
      <button
        data-testid="increment"
        type="button"
      >
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`the product cards should render correctly when cardStyle is sold-out 1`] = `
<DocumentFragment>
  <div
    class="product-card-sold-out"
    data-testid="product"
  >
    <img />
    <p />
    <p>
      MRP  /
    </p>
    <div
      data-testid="counter-div"
    >
      <button
        data-testid="decrement"
        type="button"
      >
        -
      </button>
      <button
        data-testid="increment"
        type="button"
      >
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`the product component should render correctly when cardStyle is normal 1`] = `
<DocumentFragment>
  <div
    class="product-card"
    data-testid="product"
  >
    <img />
    <p />
    <p>
      MRP  /
    </p>
    <div
      data-testid="counter-div"
    >
      <button
        data-testid="decrement"
        type="button"
      >
        -
      </button>
      <button
        data-testid="increment"
        type="button"
      >
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`the product component should render correctly when cardStyle is sold-out 1`] = `
<DocumentFragment>
  <div
    class="product-card-sold-out"
    data-testid="product"
  >
    <img />
    <p />
    <p>
      MRP  /
    </p>
    <div
      data-testid="counter-div"
    >
      <button
        data-testid="decrement"
        type="button"
      >
        -
      </button>
      <button
        data-testid="increment"
        type="button"
      >
        +
      </button>
    </div>
  </div>
</DocumentFragment>
`;
